src/unitxt/splitters.py

```diff
@@ -151,11 +146,74 @@ def filter_source_by_instance(
 
 
 class RandomSampler(Sampler):
+    """Selects a random sample of instances."""
+
     def sample(
-        self, instances_pool: List[Dict[str, object]]
+        self,
+        instances_pool: List[Dict[str, object]],
+        instance: Optional[Dict[str, object]] = None,
     ) -> List[Dict[str, object]]:
         instances_pool = list(instances_pool)
-        return self.random_generator.sample(instances_pool, self.sample_size)
+        random_generator = new_random_generator(sub_seed=str(instance))
```

Comment:
```suggestion
        random_generator = new_random_generator(sub_seed={**instance["inputs"], **instance["outputs"]})
```
Make it more robust to future changes in the instance structure (such as classification policy metadata etc that does not change the essence of the instance)

Reply:
Thanks.  I changed to code to use "input_fields" (references are not always available, and should not effect randomization if the input fields are the same).